packages/Config/RectorConfig.php

```diff
@@ -4,37 +4,31 @@
 
 namespace Rector\Config;
 
+use Illuminate\Container\Container;
 use Rector\Caching\Contract\ValueObject\Storage\CacheStorageInterface;
 use Rector\Core\Configuration\Option;
 use Rector\Core\Configuration\Parameter\SimpleParameterProvider;
-use Rector\Core\Configuration\ValueObjectInliner;
 use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
 use Rector\Core\Contract\Rector\NonPhpRectorInterface;
 use Rector\Core\Contract\Rector\PhpRectorInterface;
 use Rector\Core\Contract\Rector\RectorInterface;
 use Rector\Core\Exception\ShouldNotHappenException;
+use Rector\Core\NodeAnalyzer\ScopeAnalyzer;
+use Rector\Core\Rector\AbstractScopeAwareRector;
 use Rector\Core\ValueObject\PhpVersion;
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-use Symfony\Component\DependencyInjection\Loader\Configurator\ServiceConfigurator;
-use Symfony\Component\DependencyInjection\Loader\Configurator\ServicesConfigurator;
 use Webmozart\Assert\Assert;
 
 /**
  * @api
- * Same as Symfony container configurator, with patched return type for "set()" method for easier DX.
- * It is an alias for internal class that is prefixed during build, so it's basically for keeping stable public API.
  */
-final class RectorConfig extends ContainerConfigurator
+final class RectorConfig extends Container
```

Comment:
it would be great to have 
- `services()`: Container, 
- `set()`: Container, 
- `tag()`: void

Fluent call to return  for compatiblity existing code so we don't need to update documentation 

https://getrector.com/documentation/creating-a-node-visitor

Reply:
`services()`, `set()`, `tag()` should not trigger error imo, return and set and tag to new services instead :)

Reply:
`tag()` method seems exists in `Laravel` Container, so it probably just need 2 methods, something like this pseudo code:

```php
    public function services(): self
    {
        return $this;
    }

    public function set(string $serviceName): self
    {
        $this->make($serviceName);
        return $this;
    }
```

Reply:
I see :) Could you add those methods and add test case for it on main branch?

Reply:
On main branch will probably cause error as still using symfony, I will create a PR to this PR branch

Reply:
@TomasVotruba hm.., I think the `services()` and `set()` is fine to gone, as the code chain will no longer works to remember last services setted:

```php
$rectorConfig->services()->set(HelloVisitor::class)->tag(ScopeResolverNodeVisitorInterface::class);
```

so the updated documentation is better instead of adding compatiblity code to be something like:

```php
$rectorConfig->make(HelloVisitor::class);
$rectorservice->tag(HelloVisitor::class, ScopeResolverNodeVisitorInterface::class);
```